### ISSUE TITLE
"::" entry in notracking hosts file blocks URLs containing "::"; parse notracking list

### ISSUE THREAD

**Issue by Mathnerd314** [uBlockOrigin/uBlock-issues#800]
### Prerequisites

- [x] I verified that this is not a filter issue
    - Filter issues MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues)
    
- [x] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uBlockOrigin](https://old.reddit.com/r/uBlockOrigin/)
    
- [x] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
- I tried to reproduce the issue when...
    - [ ] uBlock Origin is the only extension
    - [ ] uBlock Origin with default lists/settings
    - [ ] using a new, unmodified browser profile
- [x] I am running the latest version of uBlock Origin
- [x] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### Description

uBlock [says](https://github.com/gorhill/uBlock/wiki/Does-uBlock-block-ads-or-just-hide-them%3F#hosts-file) it supports hosts files so I added https://raw.githubusercontent.com/notracking/hosts-blocklists/master/hostnames.txt. It seems to work fine except that the "::" at the beginning is interpreted as a filter and blocks some requests.

The other half of the notracking list is a Dnsmasq config file (https://github.com/notracking/hosts-blocklists/blob/master/domains.txt) with lines of the form `address=/zzlayejqlzcg.com/::` and `address=/zzzezeroe.fr/0.0.0.0`; they seem to be parsed incorrectly as URL filters.

Unlike #576, notracking is a useful list not available in any formats uBlock supports. It contains many small lists not included elsewhere and the original sources are often in CSV or other strange formats.

### A specific URL where the issue occurs

https://www.youtube.com/watch?v=spSxQjDguzM

### Steps to Reproduce

1. Blocked URL requests happen in the middle of YouTube videos like the linked URL. The playback stops and the spinner appears. The logger shows URLs from googlevideo.com are being blocked because they contain `::`.
2. URLs from the domain blocklist are not blocked

### Expected behavior:

* `::` is ignored like the 0.0.0.0 IP address.
* address= lines are parsed as hostname filters

### Actual behavior:

* `::` is interpreted as a generic filter rule.
* `address` lines are parsed as textual filters

### Your environment

* uBlock Origin version: 1.24.2
* Browser Name and version: Firefox 71.0b12
* Operating System and version: Windows 10

**Comment by uBlock-user** [comment]
`::` is not a hosts file format and shouldn't be used in a hosts file in the first place.

**Comment by uBlock-user** [comment]
> notracking is a useful list

It's overzealous list where the maintainer clearly doesn't care enough[1] to manually inspect the domains[2] before adding and collects them all via an automated script.

[1] - https://github.com/notracking/hosts-blocklists/issues/67
[2] - https://github.com/notracking/hosts-blocklists/issues/100

**Comment by gwarser** [comment]
Isn't `::` IPv6 equivalent of `0.0.0.0`? Or should it be `::/0`/`::/128` (default route / unspecified)?

Is it really useful in HOST files?

**Comment by gorhill** [comment]
uBO should discard `::` from a hosts file, my understanding is that it's the equivalent of `0000:0000:0000:0000:0000:0000:0000:0000`.

**Comment by gwarser** [comment]
Dan Pollock’s hosts file contains this:

	#<localhost>
	127.0.0.1	localhost
	127.0.0.1	localhost.localdomain
	255.255.255.255	broadcasthost
	::1		localhost
	127.0.0.1	local
	::1		ip6-localhost ip6-loopback
	fe00::0		ip6-localnet
	ff00::0		ip6-mcastprefix
	ff02::1		ip6-allnodes
	ff02::2		ip6-allrouters
	ff02::3		ip6-allhosts
	#fe80::1%lo0	localhost
	#</localhost>

**Comment by gorhill** [comment]
> Skipped by `reIsLocalhostRedirect`?

Yes, <https://regex101.com/r/Nh1V0R/1>.

**Comment by gwarser** [comment]
Just a side note: I'm not able to reproduce any issue with YT, because any URL with `::` is already blocked by `||youtube.com/api/stats/qoe?` from EasyPrivacy.

**Comment by krystian3w** [comment]
https://regex101.com/r/Nh1V0R/2

`https://who.is/whois/localhost.pl`

> `dnssec:                Unsigned`


So no possible blacklist in hosts format if when run in the feature?

**Comment by gorhill** [comment]
I don't understand what you are saying.

**Comment by krystian3w** [comment]
What happed if domain with real TLD start serve malware and Peter lowe blacklist in host format instead of ABP:

`127.0.0.1 localhost.pl` - uBO ommit and malware should works in the feature
`||localhost.pl^` - format no prefered in downloaded Peter Lowe List.

**Comment by gorhill** [comment]
I see. This should go in its own issue or else I will forget about this.
